src/Utilities/MoistThermodynamics/src/MoistThermodynamics.jl

```diff
@@ -462,49 +784,212 @@ function saturation_adjustment(e_int, ρ, q_tot, T_init = T_triple)
     return air_temperature(e_int, q_tot, q_liq, q_ice)
   end
 end
+function saturation_adjustment_q_t_θ_l(θ_liq::DT, q_tot::DT, ρ::DT, p::DT) where DT
+  T_1 = θ_liq*(p/DT(MSLP))^(DT(R_d)/DT(cp_d))
+  qv_star = saturation_shum(T_1, ρ)
+  if (q_tot <= qv_star) # If not saturated
+    return T_1
+  else  # If not saturated, iterate
+    T_2 = T_1 + (q_tot - qv_star) * latent_heat_vapor(T_1) /((1-q_tot)*DT(cp_d) + qv_star * DT(cp_v))
+    function eos_root(T, args)
+      qv_star = saturation_shum(T, ρ)
+      θ = T*(DT(MSLP) / p)^(DT(R_d)/DT(cp_d))
```

Comment:
Not consistent with our definition of dry potential temperature and liquid-ice potential temperature. Use those functions to make it consistent, rather than redefining potential temperature.

Reply:
So, parallel to the internal energy adjustment, this would be something like `roots_equation ... = liquid_ice_pottemp_sat(...) - liquid_ice_pottemp`

Reply:
I think I was able to clear this up, but I think there's still one loose end.